src/Workspaces/VisualBasic/Portable/Editing/VisualBasicImportAdder.vb

```diff
@@ -57,91 +59,139 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Editing
-            Inherits VisualBasicSyntaxRewriter
+        Private Class ConflictWalker
+            Inherits VisualBasicSyntaxWalker
 
-            Private ReadOnly _workspace As Workspace
             Private ReadOnly _cancellationToken As CancellationToken
             Private ReadOnly _model As SemanticModel
-            Private ReadOnly _namespaceMembers As HashSet(Of String)
-            Private ReadOnly _extensionMethods As HashSet(Of String)
 
-            Public Sub New(namespaceMembers As IEnumerable(Of INamespaceOrTypeSymbol), extensionMethods As IEnumerable(Of IMethodSymbol), model As SemanticModel, workspace As Workspace, cancellationToken As CancellationToken)
+            ''' <summary>
+            ''' A mapping containing the simple names And arity of all namespace members, mapped to the import that
+            ''' they're brought in by.
+            ''' </summary>
+            Private ReadOnly _namespaceMembers As MultiDictionary(Of (name As String, arity As Integer), INamespaceSymbol) =
+                New MultiDictionary(Of (name As String, arity As Integer), INamespaceSymbol)()
+
+            ''' <summary>
+            ''' A mapping containing the simple names of all extension methods, mapped to the import that they're
+            ''' brought in by.  This doesn't keep track of arity because methods can be called with type arguments.
+            ''' </summary>
+            Private ReadOnly _extensionMethods As MultiDictionary(Of String, INamespaceSymbol) =
+                New MultiDictionary(Of String, INamespaceSymbol)()
+
+            Private ReadOnly _conflictNamespaces As HashSet(Of INamespaceSymbol)
+
+            ''' <summary>
+            ''' Track if we're in an anonymous method or not.  If so, because of how the language binds lambdas and
+            ''' overloads, we'll assume any method access we see inside (instance or otherwise) could end up conflicting
+            ''' with an extension method we might pull in.
+            ''' </summary>
+            Private _inAnonymousMethod As Boolean
+
+            Public Sub New(
+                    model As SemanticModel,
+                    namespaceSymbols As ImmutableArray(Of INamespaceSymbol),
+                    conflictNamespaces As HashSet(Of INamespaceSymbol),
+                    cancellationToken As CancellationToken)
+                MyBase.New(SyntaxWalkerDepth.StructuredTrivia)
                 _model = model
-                _workspace = workspace
                 _cancellationToken = cancellationToken
-                _namespaceMembers = New HashSet(Of String)(namespaceMembers.[Select](Function(x) x.Name), CaseInsensitiveComparison.Comparer)
-                _extensionMethods = New HashSet(Of String)(extensionMethods.[Select](Function(x) x.Name), CaseInsensitiveComparison.Comparer)
+                _conflictNamespaces = conflictNamespaces
+
+                For Each ns In namespaceSymbols
+                    For Each typeOrNamespace In ns.GetMembers()
```

Comment:
What about VB modules? An Imports statement also brings into global scope member names contained in modules. I suspect this won't be tricky to fix -- it either needs another dictionary or you might be able to add them into _namespaceMembers...depends on how you need to deal with method arity and I'm not sure about the details there.

Reply:
good point!

Reply:
fixed.